lib/websocket-tracker.js

```diff
@@ -32,7 +32,10 @@ function WebSocketTracker (client, announceUrl, opts) {
   self._interval = null
 
   if (socketPool[announceUrl]) self._socket = socketPool[announceUrl]
-  else self._socket = socketPool[announceUrl] = new Socket(announceUrl)
+  else {
+      self._socket = socketPool[announceUrl] = new Socket(announceUrl)
+      self._socket.on('ready', function () { self._socket.ready = true })
```

Comment:
This line shouldn't be necessary because `simple-websocket` already does this: https://github.com/feross/simple-websocket/blob/master/index.js#L52

Try running `npm update` to get the latest version of all dependencies :)